### PR TITLE
Add application labels as meta labels

### PR DESCRIPTION
**Pull request by fabxc**

### CODE REVIEW

File: retrieval/discovery/marathon/conversion.go [hunk 1]
```diff
@@ -24,18 +22,25 @@ func AppsToTargetGroups(apps *AppList) map[string]*config.TargetGroup {
 func createTargetGroup(app *App) *config.TargetGroup {
 	var (
 		targets = targetsForApp(app)
-		source  = targetGroupName(app)
+		source  = sanitizeName(app.ID)
```
Comment (brian-brazil) [inline retrieval/discovery/marathon/conversion.go]: does this need to be sanitized?

**Comment by brian-brazil** [comment]
:+1: 

The removal of the `isValidApp` logic is a breaking change, so should be mentioned in the description.

File: retrieval/discovery/marathon/conversion.go (hunk 1 above)
Comment (fabxc) [inline retrieval/discovery/marathon/conversion.go]: True, as it's a label, no.

Comment (fabxc) [inline retrieval/discovery/marathon/conversion.go]: Source can be any string really. But the sanitization below could theoretically sanitize distinct IDs into the same source. So let's drop it.

Comment (fabxc) [inline retrieval/discovery/marathon/conversion.go]: Also removed other sanitizations as they end up in labels. Don't feel like we should alter input if not necessary.

**Comment by fabxc** [comment]
Should be we speaking of breaking changes for features that were never part of a release?
I always feel that it makes everyones life easier if we are able to tweak a new feature before the next release.

**Comment by brian-brazil** [comment]
I think it's good practice to call them out in the description, even if we don't need to put it in the release notes

**Comment by fabxc** [comment]
Fair enough, as long as we allow ourselves to break.

**Pull request merged**